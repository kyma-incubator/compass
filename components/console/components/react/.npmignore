.babelrc
.eslintrc
.github
*.DS_Store
*npm-debug.log
*yarn-error.log
coverage/
node_modules/

