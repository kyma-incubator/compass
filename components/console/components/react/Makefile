ci-pr: root-pr resolve build test
ci-master: root-master resolve build test

.PHONY: root-pr
root-pr:
	make -C "../../" ci-pr

.PHONY: root-master
root-master:
	make -C "../../" ci-master

.PHONY: resolve
resolve:
	npm install

.PHONY: build
build:
	npm run build

.PHONY: test
test:
	CI=true npm run test
	