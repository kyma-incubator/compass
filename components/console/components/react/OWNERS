reviewers:
    - default-approvers
    - Wawrzyn321
    - content-ui-approvers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-ui-approvers