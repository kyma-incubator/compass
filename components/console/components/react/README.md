# Kyma React Components

## Overview

This application is a ReactJS component library for the Kyma, based on the [component-library-starter](https://github.com/alanbsmith/component-library-starter).

You can see the components and their properties on the [Console](https://kyma-project.github.io/console) page.

## Usage

To import the library into an application, run the following command:

```sh
npm install --save @kyma-project/react-components
```

## Development

### Build

After the build, the webpack creates the `lib` folder.

Run once:

```sh
npm run build
```

Then run the `watch` script:

```sh
npm run build:watch
```

### Live preview

Use [react-styleguidist](https://github.com/styleguidist/react-styleguidist) as a development environment. It lists component propTypes and shows live, editable usage examples based on Markdown files.

To run the live preview:

```sh
npm run styleguide
```

To build sources for later publication:

```sh
npm run styleguide:build
```

### Test

**Jest Snapshots** manages all of the testing. **Jest Snapshots** places the tests in the `__tests__` folders, on the component level.

To run the tests once:

```sh
npm test
```

To run the watch script:

```sh
npm run test:watch
```

To view the coverage report: 

```sh
npm run test:coverage:report
```
