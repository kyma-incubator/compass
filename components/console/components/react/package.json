{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start": "npm run styleguide",
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "styleguide": "styleguidist server",
    "styleguide:build": "styleguidist build",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^21.2.1",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-styleguidist": "^8.0.6",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
