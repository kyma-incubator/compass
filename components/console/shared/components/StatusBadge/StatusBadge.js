import React from 'react';
import { Badge } from 'fundamental-react/Badge';
import PropTypes from 'prop-types';
import './StatusBadge.scss';
import classNames from 'classnames';
import { Tooltip } from '../Tooltip/Tooltip';

const resolveType = status => {
  if (typeof status !== 'string') {
    console.warn(
      `'autoResolveType' prop requires 'children' prop to be a string.`,
    );
    return undefined;
  }

  switch (status.toUpperCase()) {
    case 'INITIAL':
      return 'info';

    case 'READY':
    case 'RUNNING':
    case 'SUCCESS':
    case 'SUCCEEDED':
      return 'success';

    case 'UNKNOWN':
    case 'WARNING':
      return 'warning';

    case 'FAILED':
    case 'ERROR':
    case 'FAILURE':
      return 'error';

    default:
      return undefined;
  }
};

export const StatusBadge = ({
  tooltipContent,
  type,
  children: value,
  autoResolveType = false,
  tooltipProps = {},
  className,
}) => {
  if (autoResolveType) type = resolveType(value);

  const classes = classNames(
    'status-badge',
    {
      ['status-badge--' + type]: type,
      'has-tooltip': tooltipContent,
    },
    className,
  );

  const badgeElement = (
    <Badge role="status" modifier="filled" className={classes}>
      {value}
    </Badge>
  );

  return tooltipContent ? (
    <Tooltip content={tooltipContent} {...tooltipProps}>
      {badgeElement}
    </Tooltip>
  ) : (
    badgeElement
  );
};

StatusBadge.propTypes = {
  tooltipContent: PropTypes.node,
  type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  autoResolveType: PropTypes.bool,
  tooltipProps: PropTypes.object,
  className: PropTypes.string,
};
