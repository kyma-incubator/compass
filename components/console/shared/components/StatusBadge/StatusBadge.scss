.fd-badge.status-badge {
  font-size: var(--sapFontSize, 12.25px);
  font-family: var(
    --sapFontFamily,
    '72',
    '72full',
    Arial,
    Helvetica,
    sans-serif
  );
  font-weight: 700;
  line-height: 1;
  word-break: keep-all;
  text-transform: uppercase;

  margin: 0;
  padding: 0.14rem 0.4rem;

  display: inline-block;
  max-width: 100%;
  height: auto;
  min-height: 1rem;
  width: auto;
  border-radius: 0.125rem;

  background-color: var(--sapButton_Neutral_Background, #6a6d70);
  color: var(--sapButton_Neutral_TextColor, #fff);

  &--success {
    background-color: var(--sapButton_Success_Background, #107e3e);
  }
  &--warning {
    background-color: var(--sapButton_Critical_Background, #e9730c);
  }
  &--error {
    background-color: var(--sapButton_Negative_Background, #b00);
  }
  &--info {
    background-color: var(--sapInformativeElementColor, #0a6ed1);
  }

  &.has-tooltip {
    cursor: help;
  }
}
