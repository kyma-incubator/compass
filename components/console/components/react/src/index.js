import components from './components';
import commons from './commons';

module.exports = {
  ...components,
  ...commons,
};
