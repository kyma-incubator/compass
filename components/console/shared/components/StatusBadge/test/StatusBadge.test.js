import React from 'react';
import { StatusBadge } from '../StatusBadge';
import { render } from '@testing-library/react';

describe('StatusBadge', () => {
  it('renders status text with proper role', () => {
    const { queryByRole } = render(<StatusBadge>INITIAL</StatusBadge>);

    const status = queryByRole('status');
    expect(status).toBeInTheDocument();
    expect(status).toHaveTextContent('INITIAL');
  });

  it('displays warning when autoResolveType is set and "children" is a node', () => {
    console.warn = jest.fn();

    render(
      <StatusBadge autoResolveType>
        <small>Status</small>
      </StatusBadge>,
    );

    expect(console.warn.mock.calls[0]).toMatchSnapshot();
  });
});
