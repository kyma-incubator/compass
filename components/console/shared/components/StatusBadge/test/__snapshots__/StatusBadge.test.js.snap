// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBadge displays warning when autoResolveType is set and "children" is a node 1`] = `
Array [
  "'autoResolveType' prop requires 'children' prop to be a string.",
]
`;
